### PR TITLE
Feature flag system

### LINKED ISSUE #564: Add feature flag system
Create a JSON object somewhere in the code that holds feature flags like so:
```json
{
  "chapter_1_enabled": true,
  "chapter_2_enabled": true,
  "chapter_3_enabled": false,
  ...
}
```

Based on these feature flags the lesson content should be disabled and it should become impossible to navigate to the lesson.
After https://github.com/saving-satoshi/saving-satoshi-backend/issues/20 has been implemented we can replace the json object with the response from `/v1/features`.

### PR DESCRIPTION
**Pull request by afterburn**
Closes #564

https://github.com/saving-satoshi/saving-satoshi-backend/pull/22 should be merged before this PR

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
hmmm ran into an issue. I completed ch. 1 but am unable to start ch. 2. 

https://github.com/saving-satoshi/saving-satoshi/assets/32662508/a7ccab68-00fb-4dd0-98ae-7de5ccb5b163

File: providers/FeatureProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  isFeatureEnabled: (key: string) => false,
+}
+
+export const FeatureContext = createContext<FeatureContextType>(
+  defaultFeatureContext
+)
+
+export const useFeatureContext = () => useContext(FeatureContext)
+
+export default function ProgressProvider({
```
Comment (benalleng) [inline providers/FeatureProvider.tsx:20]: I think this should be FeatureProvider()

**Review by benalleng — CHANGES_REQUESTED** [review]
One small thing otherwise Looks good!

**Comment by benalleng** [comment]
@ecurrencyhodler just tested this again and going from Chapter 1 to Chapter 2

**Comment by ecurrencyhodler** [comment]
Ditto. it works!

**Pull request merged**